lib/csv/row.rb

```diff
@@ -29,7 +29,7 @@ class Row
     #
     def initialize(headers, fields, header_row = false)
       @header_row = header_row
-      headers.each { |h| h.freeze if h.is_a? String }
+      headers = headers.map { |h| h.is_a?(String) ? -h : h }
```

Comment:
Is this change necessary?

Reply:
It's because my change broke the following test in `test_row.rb`

```
  def test_to_hash
    hash = @row.to_hash
    assert_equal({"A" => @row["A"], "B" => @row["B"], "C" => @row["C"]}, hash)
    hash.keys.each_with_index do |string_key, h|
      assert_predicate(string_key, :frozen?)
      assert_same(string_key, @row.headers[h])
    end
  end
```

The assert_same was failing because it was relying on the header strings to be used directly by the hash. Maybe it's the test I should change not, sure.